core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorTableHandle.java

```diff
@@ -13,10 +13,29 @@
  */
 package io.trino.spi.connector;
 
+import java.util.Random;
+
 /**
  * Represents a handle to a relation returned from the connector to the engine.
  * It will be used by the engine whenever given relation will be accessed.
  */
 public interface ConnectorTableHandle
 {
+    /**
+     * Returns true if a ConnectorTableHandle is "fusible" with another ConnectorTableHandler.
+     * Fusible here means the two tables can possibly be optimized by SharedScanOptimizer.
```

Comment:
Please clarify exactly what this means for connector implementors. They shouldn't have to read and understand a ~1600 line optimizer to know how to implement this method.